### ISSUE TITLE
Should we stop limiting ngd publications?

### ISSUE THREAD

**Issue by finnagin** [RTXteam/RTX#1826]
There have been a few instances where specific pmids that users were looking for to validate results were not showing up that were actually just cut off because of the 30 pmid publication limit on ngd edges. I was thinking that maybe what we should be doing is instead of limiting the list that is returned in the attribute is maybe we could just limit the number of pmids shown in the UI. Any thoughts on this @isbluis ?

**Comment by finnagin** [comment]
One potential problem that I see is that some of the responses could get pretty large so the dev info json could get very hard to read is a few results have a ton of supporting pmids.

**Comment by edeutsch** [comment]
I do still think there should be a limit. But 30 may be a bit too low. Maybe 100 is still manageable. 

I once heard a suggestions (from Marc I think) that ranking PMIDs by their number of citations would be cool. and I agree. I wonder if we have any reasonable chance of gathering the number of citations for all these papers? Probably not, this is closely guarded I think. Is there any kind of ranking? maybe at least reverse integer sorting the list to get the most recent papers first would be a good start?

Another fun idea I had was: wouldn't it be cool if you could ask RTX-KG2 "what assertions do you have that reference PMID 123456789 as evidence?" It that something that would be feasible/interesting? I wonder how many assertions had a link to a paper? I think SemMedDB does, but probably not many others?

**Comment by amykglen** [comment]
not sure if that info is in the pubmed dump (worth checking), but I wonder if the Google Scholar API could be used to do it?

> Another fun idea I had was: wouldn't it be cool if you could ask RTX-KG2 "what assertions do you have that reference PMID 123456789 as evidence?" It that something that would be feasible/interesting? I wonder how many assertions had a link to a paper? I think SemMedDB does, but probably not many others?

was wondering - could that theoretically be done using constraints in a TRAPI query? like, for a one-hop TRAPI query, add a constraint to the qedge requiring that PMIDXXXX is in that edge's publications attribute? (of course I doubt anyone supports this at the moment, but curious if that's a valid use of constraints..) I suppose that would still be subject to our typical requirement that at least one qnode in a query is pinned though.

**Comment by edeutsch** [comment]
Yes, I think that would be a valid constraint. It would be groovy if Expand() could support that.

**Comment by edeutsch** [comment]
Ah, but you're right that >=1 pinned node is a requirement that precludes my exact use case.

**Comment by finnagin** [comment]
Posted @dkoslicki 's message from slack here:
Re: finding citations, besides the PMC approach (https://www.ncbi.nlm.nih.gov/pmc/tools/cites-citedby/)
There appears to be a way to get Google citations for PMIDs via the following: PMID->DOI (https://www.ncbi.nlm.nih.gov/pmc/tools/idconv/result/), DOI->Google CID (https://scholar.google.com/scholar?hl=en&as_sdt=0%2C39&q=10.3410%2FB3-15&btnG= <-- search term using the DOI), Google CID -> Citation count (https://scholar.google.com/scholar?cites=10134813737474265221&as_sdt=5,39&sciodt=0,39&hl=en <-- search using the cid returned from the previous search)
Seems tedious to chain it together though

**Comment by finnagin** [comment]
Didn't end up having time for this one before I left.

Current plan: Run through pubmed xml dump and rank the pmids in the ngd pmid database in order of citations. Assigning @edeutsch and @amykglen to workout the details and implement at a later date.

**Comment by edeutsch** [comment]
I think this issue has drifted into the "interesting ideas, but not really an issue that is impacting anyone". Okay to close?

**Comment by dkoslicki** [comment]
Agreed, fine by me to close
